Library/Homebrew/test/rubocops/text_spec.rb

```diff
@@ -226,5 +226,30 @@ def install
         end
       RUBY
     end
+
+    it "When a single inreplace is in block form" do
+      expect_offense(<<~RUBY)
+        class Foo < Formula
+          def install
+            inreplace "foo" do |s|
+            ^^^^^^^^^^^^^^^^^^^^^^ `inreplace ... do` was used for a single substitution (use the non-block form instead).
+              s.gsub! "bar", "baz"
+            end
+          end
+        end
+      RUBY
+    end
+
+    it "When a single inreplace would exceed the line length limit" do
+      expect_no_offenses(<<~RUBY)
+        class Foo < Formula
+          def install
+            inreplace "this_is_a_really_long_file_name.txt" do |s|
+              s.gsub! "this is a really really long string", "this is another very long string"
+            end
+          end
+        end
+      RUBY
+    end
```

Comment:
This test fails currently.

The issue I'm seeing with this `inreplace` check is that there are some of the homebrew/core formulae use the block form because the one-line form is too long. Is this acceptable? If so, how should I check for that with RuboCop?

Reply:
> The issue I'm seeing with this `inreplace` check is that there are some of the homebrew/core formulae use the block form because the one-line form is too long. Is this acceptable?

I think so!



> If so, how should I check for that with RuboCop?

I'm not sure. RuboCop will enforce the line length for us so I wonder if we can just be more tolerant with this check?

Reply:
I looked into approximating what the line length would be if the block was converted to a one-line `inreplace` by adding the length of the strings to the length of the word `inreplace` and adding some spaces and quotes. Something like this:

```ruby
count = 'inreplace "", "", ""'.length # Length of keywords, commas, and quotes
find_strings(b).each { |s| count += s.source.length }

next if count > 118
```

It felt a little too unreliable/hacky to me, but maybe its reasonable.

here's another possible solution that might be better:

```ruby
next unless match = b.source.match(/inreplace ([^\n]*) do [^\n]*\n[^\n]*\.gsub!([^\n]*\n\ *)end/m)

next if "inreplace #{match[1]}, #{match[2]}".length > 118
```

If RuboCop doesn't have a way to easily determine whether one line would be too long, should I remove the check altogether? I can't think of how to handle it best.

Either way, it looks like the old audit rule didn't always work as there are some formulae in homebrew/core that don't follow the rule even though they could...

Reply:
> If RuboCop doesn't have a way to easily determine whether one line would be too long, should I remove the check altogether? I can't think of how to handle it best.

Yeh, I think just remove this 👍

Reply:
Should I also remove it from `audit`? (If so, I'll remove the `#TODO:` comment)

Reply:
Yes please!

Reply:
Done!